pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -144,28 +140,24 @@ func (r *reconciler) currentLoadBalancerService(ci *operatorv1.IngressController
 // that's no longer necessary. We just need to clear the finalizer which might exist
 // on existing resources.
 // TODO: How can we delete this code?
-func (r *reconciler) finalizeLoadBalancerService(ci *operatorv1.IngressController) error {
+func (r *reconciler) finalizeLoadBalancerService(ci *operatorv1.IngressController) (*corev1.Service, error) {
```

Comment:
Instead of a pointer, how about returning a Boolean value to indicate presence of the service?